src/components/legend/index.js

```diff
@@ -435,22 +443,36 @@ legend.draw = function(td) {
 
     var legendsvg = fullLayout._infolayer.selectAll('svg.legend')
         .data([0]);
-    legendsvg.enter(0).append('svg')
-        .attr('class','legend');
+    legendsvg.enter().append('svg')
+        .attr('class','legend')
+        .attr('pointer-events', 'all');
 
-    var bgRect = legendsvg.selectAll('rect.bg')
+    var bg = legendsvg.selectAll('rect.bg')
         .data([0]);
-    bgRect.enter(0).append('rect')
-        .attr('class','bg');
-    bgRect
-        .call(Plotly.Color.stroke, opts.bordercolor)
-        .call(Plotly.Color.fill, opts.bgcolor)
-        .style('stroke-width', opts.borderwidth+'px');
-
-    var groups = legendsvg.selectAll('g.groups')
-        .data(legendData);
+    bg.enter().append('rect')
+        .attr('class','bg')
+        .call(Color.stroke, opts.bordercolor)
+        .call(Color.fill, opts.bgcolor)
+        .style('stroke-width', opts.borderwidth + 'px');
+
+    var scrollBox = legendsvg.selectAll('svg.scrollbox')
+        .data([0]);
+    scrollBox.enter().append('svg')
+        .attr('class', 'scrollbox');
+    scrollBox.exit().remove();
 
-    groups.enter().append('g').attr('class', 'groups');
+    var scrollBar = legendsvg.selectAll('rect.scrollbar')
+        .data([0]);
+    scrollBar.enter().append('rect')
```

Comment:
looks like all the failed image tests are caused by this scroll bar `rect`, e.g

![image](https://cloud.githubusercontent.com/assets/6675409/12921033/5385c280-cf19-11e5-8b27-e3c4bc797024.png)

We should add logic here so that the scroll bar only appears when the legend is in fact scrollable.

In addition, we should remove the scroll bar - no matter the size of legend - on image exports.